src/smart-components/workspaces/WorkspaceDetail.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchWorkspace, fetchWorkspaces } from '../../redux/actions/workspaces-actions';
+import { ContentHeader } from '@patternfly/react-component-groups';
+import { Breadcrumb, BreadcrumbItem, Divider, PageSection, Text, TextContent, TextVariants } from '@patternfly/react-core';
+import { RBACStore } from '../../redux/store';
+import { useParams } from 'react-router-dom';
+import AppTabs from '../app-tabs/app-tabs';
+import pathnames from '../../utilities/pathnames';
+import WorkspaceActions from './WorkspaceActions';
+import Messages from '../../Messages';
+
+interface WorkspaceData {
+  name: string;
+  uuid: string;
+}
+
+const WorkspaceDetail = () => {
+  const intl = useIntl();
+  const { workspaceId } = useParams();
+
+  const dispatch = useDispatch();
+
+  const { isLoading, workspaces, selectedWorkspace, error } = useSelector((state: RBACStore) => state.workspacesReducer);
+  const [workspaceHierarchy, setWorkspaceHierarchy] = useState<WorkspaceData[]>([]);
+
+  useEffect(() => {
+    if (workspaceId && workspaceId.length > 0) {
+      dispatch(fetchWorkspace(workspaceId));
+    }
+    dispatch(fetchWorkspaces());
+  }, [dispatch]);
+
+  const buildWorkspacesHierarchy = (workspaces: any[], workspaceId: string): WorkspaceData[] => {
+    let currentWorkspace = workspaces.find((ws) => ws.uuid === workspaceId);
```

Comment:
you can probably use `selectedWorkspace`

Reply:
I'm not sure if I can do that assuming that selectedWorkspace is mainly used to get workspace name and description in the header and we don't want that to be changed after the page gets loaded.